src/bp-activity/bp-activity-functions.php

```diff
@@ -2296,45 +2306,94 @@ function bp_activity_post_update( $args = '' ) {
-			'hide_sitewide' => $r['hide_sitewide'],
-			'privacy'       => $r['privacy'],
-			'error_type'    => $r['error_type'],
-		)
-	);
+	if ( ! empty( $r['id'] ) ) {
+		$activity = new BP_Activity_Activity( $r['id'] );
+
+		if ( ! empty( $activity->id ) ) {
+			$bp_activity_edit = true;
+
+			$activity_id = bp_activity_add(
+				array(
+					'id'                => $activity->id,
+					'action'            => $activity->action,
+					'content'           => $add_content,
+					'component'         => $activity->component,
+					'type'              => $activity->type,
+					'primary_link'      => $add_primary_link,
+					'user_id'           => $activity->user_id,
+					'item_id'           => $activity->item_id,
+					'secondary_item_id' => $activity->secondary_item_id,
+					'recorded_time'     => $activity->date_recorded,
+					'hide_sitewide'     => $activity->hide_sitewide,
+					'is_spam'           => $activity->is_spam,
+					'privacy'           => $r['privacy'],
+					'error_type'        => $r[ 'error_type' ],
+				)
+			);
+
+			/**
+			 * Addition from the BuddyBoss
+			 * Add meta to ensure that this activity has been edited.
+			 */
+
+			bp_activity_update_meta( $activity->id, '_is_edited', current_time( 'mysql' ) );
+
+		}
+	} else {
+		// Now write the values.
+		$activity_id = bp_activity_add(
+			array(
+				'user_id'       => $r['user_id'],
+				'content'       => $add_content,
+				'primary_link'  => $add_primary_link,
+				'component'     => $r['component'],
+				'type'          => $r['type'],
+				'hide_sitewide' => $r['hide_sitewide'],
+				'privacy'       => $r['privacy'],
+				'error_type'    => $r['error_type'],
+			)
+		);
+	}
 
 	// Bail on failure.
 	if ( false === $activity_id || is_wp_error( $activity_id ) ) {
 		return $activity_id;
 	}
 
 	if ( ! empty( $r['content'] ) && ! strlen( trim( $r['content'] ) ) ) {
-		/**
-		 * Filters the latest update content for the activity item.
-		 *
-		 * @since BuddyPress 1.6.0
-		 *
-		 * @param string $r Content of the activity update.
-		 * @param string $activity_content Content of the activity update.
-		 */
-		$activity_content = apply_filters( 'bp_activity_latest_update_content', $r['content'], $activity_content );
+		$update_activity = true;
 
-		// Add this update to the "latest update" usermeta so it can be fetched anywhere.
-		bp_update_user_meta(
-			bp_loggedin_user_id(),
-			'bp_latest_update',
-			array(
-				'id'      => $activity_id,
-				'content' => $activity_content,
-			)
-		);
+		if ( $bp_activity_edit ) {
+			$latest_activity = bp_get_user_meta( bp_loggedin_user_id(), 'bp_latest_update', true );
+
+			if ( $latest_activity['id'] !== $activity_id ) {
+				$update_activity = false;
+			}
+		}
+
+		if ( $update_activity ) {
+
+			/**
+			 * Filters the latest update content for the activity item.
+			 *
+			 * @param string $r Content of the activity update.
+			 * @param string $activity_content Content of the activity update.
+			 *
+			 * @since BuddyPress 1.6.0
```

Comment:
@shohel Make the since tag proper every new functions and newly added filter or action